extensions/binding/org.eclipse.smarthome.binding.tradfri/src/main/java/org/eclipse/smarthome/binding/tradfri/handler/TradfriGatewayHandler.java

```diff
@@ -114,6 +131,43 @@ public void initialize() {
         }, 0, 1, TimeUnit.MINUTES);
     }
 
+    protected GatewayConfig getUsernameAndKey(GatewayConfig configuration) {
+        try {
+            DtlsConnectorConfig.Builder builder = new DtlsConnectorConfig.Builder(new InetSocketAddress(0));
+            builder.setPskStore(new StaticPskStore("Client_identity", configuration.code.getBytes()));
+
+            DTLSConnector dtlsConnector = new DTLSConnector(builder.build());
+            CoapEndpoint authEndpoint = new CoapEndpoint(dtlsConnector, NetworkConfig.getStandard());
+            String authUrl = "coaps://" + configuration.host + ":" + configuration.port + "/15011/9063";
```

Comment:
Will this work on older firmwares? If not, it would be a good moment to handle that case here.

Reply:
If you'r gateway is still not updated - could you please check if such authentication works on it:
```builder.setPskStore(new StaticPskStore("identity", configuration.code.getBytes()));```
Or it requires the ```identity``` to be blank ?

(If it works - then older firmware handling will be a lot easier :)

Reply:
Unfortunately, my firmware is already updated, so I cannot test. But my assumption is that it simply does not support this authentication mechanism as this has only been introduced in the latest firmware. So checking if the firmware is at least a certain version and going to OFFLINE with some message otherwise would imho make sense.

Reply:
So you might want to use your method `isOldFirmware` here as well to make sure that you are not creating this request on older gateways.